### PR TITLE
Add `GithubObject.last_modified_datetime` to have `last_modified` as a `datetime`

### PR DESCRIPTION
**Pull request by chouetz**
As nearly all dates are returned as `datetime.datetime`, it would be convenient to have `last_modified` as a `datetime.datetime` as well.
As discussed below, let's introduce `last_modified_datetime` new argument to cover this need.

### CODE REVIEW

**Comment by EnricoMi** [comment]
This would be a breaking change, and `last_modified` could rightly be considered a header attribute of type string,

I'd prefer to introduce a separate property `last_modified_datetime: Optional[datetime]`. Using `_makeDatetimeAttribute` would also help to support more than just a single hard-coded format.

**Comment by chouetz** [comment]
> This would be a breaking change, and `last_modified` could rightly be considered a header attribute of type string,
> 
> I'd prefer to introduce a separate property `last_modified_datetime: Optional[datetime]`. Using `_makeDatetimeAttribute` would also help to support more than just a single hard-coded format.
> 
> And here should be some asserts on this property in tests.

Many thanks for your comment, I'll update the PR this way and let you know when it's ready

**Comment by chouetz** [comment]
Probably needs #2783 for the failing job, @EnricoMi can you check the PR content again please?
Many thanks in advance

File: github/GithubObject.py [hunk 1]
```diff
@@ -302,6 +302,13 @@ def last_modified(self) -> Optional[str]:
         """
         return self._headers.get(Consts.RES_LAST_MODIFIED)  # type: ignore
 
+    @property
+    def last_modified_datetime(self) -> Optional[datetime]:
+        """
+        :type: datetime
+        """
+        return self._makeDatetimeAttribute(self._headers.get(Consts.RES_LAST_MODIFIED))  # type: ignore
```
Comment (EnricoMi) [inline github/GithubObject.py:310]: We can simplify this to
```suggestion
        return self._makeDatetimeAttribute(self.last_modified)  # type: ignore
```

Comment (EnricoMi) [reply github/GithubObject.py:310]: Method `self._makeDatetimeAttribute` returns an `Attribute`, but we want to return the value, so we need to
```suggestion
        return self._makeDatetimeAttribute(self._headers.get(Consts.RES_LAST_MODIFIED)).value  # type: ignore
```

**Review by EnricoMi — CHANGES_REQUESTED** [review]

File: github/GithubObject.py [hunk 2]
```diff
@@ -302,6 +302,13 @@ def last_modified(self) -> Optional[str]:
         """
         return self._headers.get(Consts.RES_LAST_MODIFIED)  # type: ignore
 
+    @property
+    def last_modified_datetime(self) -> Optional[datetime]:
+        """
+        :type: datetime
+        """
+        return self._makeDatetimeAttribute(self.last_modified)  # type: ignore
```
Comment (chouetz) [inline github/GithubObject.py:310]:
```suggestion
        return self._makeDatetimeAttribute(self.last_modified).value  # type: ignore
```

**Comment by chouetz** [comment]
Thanks for your time @EnricoMi

**Comment by chouetz** [comment]
OK for review @EnricoMi, many thanks in advance

**Review by EnricoMi — APPROVED** [review]
LGTM!

**Pull request merged**